optuna/samplers/nsgaii/_sampler.py

```diff
@@ -345,6 +345,12 @@ def _fast_non_dominated_sort(
         population: List[FrozenTrial],
         directions: List[optuna.study.StudyDirection],
     ) -> List[List[FrozenTrial]]:
+        if self._constraints_func is not None:
+            for _trial in population:
+                _constraints = _trial.system_attrs.get(_CONSTRAINTS_KEY)
+                if np.any(np.isnan(_constraints)):
+                    raise ValueError("NaN is not acceptable as constraint value.")
```

Comment:
Hmm, we got a mypy error here.
https://github.com/optuna/optuna/actions/runs/3402663220/jobs/5658649395

```
optuna/samplers/nsgaii/_sampler.py:351: error: Argument 1 to "__call__" of "_UFunc_Nin1_Nout1" has incompatible type "Optional[Any]"; expected "Union[int, float, complex, str, bytes, generic]"
Found 1 error in 1 file (checked 296 source files)
Error: Process completed with exit code 1.
```

Reply:
Do we need to check if `_constraints is not None`?

Reply:
I added it!